enterprise/kernel/src/main/java/org/neo4j/kernel/enterprise/builtinprocs/EnterpriseBuiltInDbmsProcedures.java

```diff
@@ -83,20 +80,20 @@
     public void setTXMetaData( @Name( value = "data" ) Map<String,Object> data )
     {
         securityContext.assertCredentialsNotExpired();
-        int totalCharSize = data.entrySet().stream()
-                .mapToInt( e -> e.getKey().length() + e.getValue().toString().length() )
-                .sum();
-
-        if ( totalCharSize >= HARD_CHAR_LIMIT )
```

Comment:
Why do we remove this limit? I am guessing it was there to prevent a kind of DoS, whereby injecting massive maps would slow the server down, since the maps are used in query logging and query listing.